pyopencl/tools.py

```diff
@@ -1234,38 +1285,48 @@ def dtypes_match():
 
 # {{{ code generation/templating helper
 
-def _process_code_for_macro(code):
+def _process_code_for_macro(code: str) -> str:
     code = code.replace("//CL//", "\n")
 
     if "//" in code:
-        raise RuntimeError("end-of-line comments ('//') may not be used in "
-                "code snippets")
+        raise RuntimeError(
+            "end-of-line comments ('//') may not be used in code snippets")
 
     return code.replace("\n", " \\\n")
 
 
-class _SimpleTextTemplate:
-    def __init__(self, txt):
-        self.txt = txt
+class _TextTemplate(ABC):
```

Comment:
The subclasses of this could be dataclasses it seems.

Reply:
Made them into dataclasses in https://github.com/inducer/pyopencl/compare/3d7d45e82efd6e92488dad748cf9781e27b32a00..bfc1af2939357dd2ef6ae4dbe45bfd4869f29412. The `mako` wrapper was a bit awkward, but it looks fine.